Controllers/HomeController.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Zoltu.Bags.Client.Controllers
+{
+	public class HomeController : Controller
+	{
+		[Route("")]
+		[Route("app")]
+		[Route("app/{path}")]
+		[Route("app/product/{productId}")]
+		public IActionResult App(int productId = 0, string path = null)

```

Comment:
Once you remove the route with `productId` in it, this parameter (`productId`) can then be removed.